source/Halibut.Tests/ServiceModel/PendingRequestQueueFixture.cs

```diff
@@ -99,27 +99,31 @@ public async Task QueueAndWait_WhenPollingRequestQueueTimeoutIsReached_WillStopW
             response.Id.Should().Be(request.Id);
             response.Error.Message.Should().Be("A request was sent to a polling endpoint, but the polling endpoint did not collect the request within the allowed time (00:00:01), so the request timed out.");
 
-            var next = await sut.DequeueAsync();
+            var next = await sut.DequeueAsync(CancellationToken);
             next.Should().BeNull();
         }
 
         [Test]
-        [TestCaseSource(typeof(AsyncTestCase))]
-        public async Task QueueAndWait_WhenRequestIsDequeued_ButPollingRequestQueueTimeoutIsReached_AndPollingRequestMaximumMessageProcessingTimeoutIsReached_WillStopWaitingAndClearRequest(bool async)
+        [SyncAndAsync]
+        public async Task QueueAndWait_WhenRequestIsDequeued_ButPollingRequestQueueTimeoutIsReached_AndPollingRequestMaximumMessageProcessingTimeoutIsReached_WillStopWaitingAndClearRequest(SyncOrAsync syncOrAsync)
         {
             // Arrange
             const string endpoint = "poll://endpoint001";
 
-            var sut = new PendingRequestQueueBuilder().WithEndpoint(endpoint).Build();
+            var sut = new PendingRequestQueueBuilder()
+                .WithAsync(syncOrAsync)
+                .WithEndpoint(endpoint)
+                .WithPollingQueueWaitTimeout(TimeSpan.Zero) // Remove delay, otherwise we wait the full 20 seconds for DequeueAsync at the end of the test
```

Comment:
Turns out the "Set" bug was allowing some of these tests to finish quickly and ignore the "Delay" (as they weren't waiting like they should). 

It's the `DequeueAsync` at the end that caused the delay.

So when we fixed it, these started taking 20 seconds to finish... These tests are not exercising that logic, so make the delay 0 so the tests do not take longer than they should.